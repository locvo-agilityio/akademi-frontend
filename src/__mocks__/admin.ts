export const MOCK_ADMIN = {
  id: 1,
  name: 'Nabila A',
  role: 'Admin',
  avatar: 'https://bit.ly/dan-abramov',
};
