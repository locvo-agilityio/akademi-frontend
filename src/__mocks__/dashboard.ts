import {
  CalendarIcon,
  FoodIcon,
  StudentIcon,
  TeacherIcon,
} from '@/components/icons';

export const MOCK_SUMMARY_DASHBOARD = [
  {
    title: 'Students',
    total: 932,
    bgIcon: 'primary',
    icon: StudentIcon,
  },
  {
    title: 'Teachers',
    total: 754,
    bgIcon: 'secondary',
    icon: TeacherIcon,
  },
  {
    title: 'Events',
    total: 40,
    bgIcon: 'yellow',
    icon: CalendarIcon,
  },
  {
    title: 'Foods',
    total: 32,
    bgIcon: 'darkBlue',
    icon: FoodIcon,
  },
];
