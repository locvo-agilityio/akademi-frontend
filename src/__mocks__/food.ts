export const MOCK_FOODS = [
  {
    name: 'Beef Steak with Fried Potato',
    content: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit.',
    image:
      'https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcRCQPUcFz7CxVScB-zxL5pl2w6f4ntjw4mV-w&s',
  },
  {
    name: 'Beef Steak with Fried Potato New',
    content: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit.',
    image:
      'https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcRCQPUcFz7CxVScB-zxL5pl2w6f4ntjw4mV-w&s',
  },
];
