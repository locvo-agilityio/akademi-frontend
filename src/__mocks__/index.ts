export * from './admin';
export * from './dashboard';
export * from './student';
export * from './pagination';
export * from './message';
export * from './food';
export * from './teacher';
export * from './schedule';
export * from './payment';
