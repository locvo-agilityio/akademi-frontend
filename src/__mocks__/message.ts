export const MOCK_MESSAGES = [
  {
    name: 'Samantha William',
    content: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit.',
    avatar: 'https://bit.ly/dan-abramov',
    time: '12:45 PM',
  },
  {
    name: 'Tony Soap',
    content: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit.',
    avatar: 'https://bit.ly/dan-abramov',
    time: '12:45 PM',
  },
  {
    name: 'Jordan Nico',
    content: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit.',
    avatar: 'https://bit.ly/dan-abramov',
    time: '12:45 PM',
  },
  {
    name: 'Nadila Adja',
    content: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit.',
    avatar: 'https://bit.ly/dan-abramov',
    time: '12:45 PM',
  },
];
