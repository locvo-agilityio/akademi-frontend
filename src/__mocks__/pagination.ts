export const MOCK_PAGINATION = {
  totalRecords: '100 items',
  pageSize: 10,
  currentButtons: [1, 2, 3],
  currentPage: 1,
};
