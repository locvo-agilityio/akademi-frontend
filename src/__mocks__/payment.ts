export const MOCK_PAYMENTS = [
  {
    id: 1,
    date: String(new Date()),
    amount: 100000,
    status: 'Pending',
  },
  {
    id: 2,
    date: String(new Date()),
    amount: 100000,
    status: 'Complete',
  },
  {
    id: 3,
    date: String(new Date()),
    amount: 100000,
    status: 'Canceled',
  },
  {
    id: 4,
    date: String(new Date()),
    amount: 100000,
    status: 'Complete',
  },
];
