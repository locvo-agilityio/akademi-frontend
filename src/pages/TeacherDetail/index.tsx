'use client';

import { HStack } from '@chakra-ui/react';
import { useParams } from 'react-router-dom';
import { lazy, Suspense } from 'react';

// Hooks
import { useGetTeacher } from '@/hooks';

// Components
import { Fallback } from '@/components';

const ProfileTeacher = lazy(() => import('@/ui/TeacherDetail/Profile'));
const ScheduleTeacher = lazy(() => import('@/ui/TeacherDetail/Schedule'));

const TeacherDetail = () => {
  const id = useParams().id || '';
  const { teacher } = useGetTeacher(id);
  const {
    address = '',
    phone = '',
    email = '',
    photo = '',
    firstName = '',
    lastName = '',
    subject = '',
    degree = '',
    description = '',
    startEducation = '',
    endEducation = '',
    university = '',
    schedule = [],
    expertise = '',
  } = teacher?.data || {};

  return (
    <HStack
      gap={10}
      mt={5}
      alignItems="flex-start"
      justifyContent="space-between"
    >
      <Suspense fallback={<Fallback />}>
        <ProfileTeacher
          address={address}
          phone={phone}
          email={email}
          avatar={photo}
          name={`${firstName} ${lastName}`}
          subject={subject}
          degree={degree}
          description={description}
          startEducation={startEducation}
          endEducation={endEducation}
          university={university}
          expertise={expertise}
        />

        <ScheduleTeacher schedules={schedule} />
      </Suspense>
    </HStack>
  );
};

export default TeacherDetail;
