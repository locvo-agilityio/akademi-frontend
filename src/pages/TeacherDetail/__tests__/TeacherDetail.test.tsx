import { render } from '@testing-library/react';

// Components
import TeacherDetail from '..';

// Mocks
import { MOCK_TEACHERS } from '@/__mocks__';

jest.mock('@/hooks', () => ({
  useGetTeacher: jest.fn(() => ({
    teacher: MOCK_TEACHERS[0],
  })),
}));

describe('TeacherDetail', () => {
  it('should render successfully', () => {
    const { container } = render(<TeacherDetail />);

    expect(container).toMatchSnapshot();
  });
});
