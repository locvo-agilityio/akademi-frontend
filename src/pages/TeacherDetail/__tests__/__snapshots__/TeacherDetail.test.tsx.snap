// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TeacherDetail should render successfully 1`] = `
<div>
  <div
    class="chakra-stack css-vwmgls"
  >
    <div
      class="css-tkevr6"
    >
      <div
        class="chakra-spinner css-1y7joxr"
        title="Fallback"
      >
        <span
          class="css-8b45rq"
        >
          Loading...
        </span>
      </div>
    </div>
  </div>
</div>
`;
